pages/api/v1/answers.ts

```diff
@@ -774,3 +801,123 @@ async function storeActionsAwaitingConfirmation(
     await redis.expire(redisKey, 60 * 10);
   }
 }
+
+async function filterActions(
+  actions: ActionPlusApiInfo[],
+  conversationId: number,
+  userQuery: string,
+  model: string,
+): Promise<ActionPlusApiInfo[]> {
+  /**
+   * Only include actions that have either:
+   * Been used previously in the conversation
+   * or
+   * Are relevant to the user's query
+   */
+
+  const actionsUsedRedisKey = conversationId.toString() + "-actions-used";
+  const actionsUsedStr = redis
+    ? ((await redis.get(actionsUsedRedisKey)) as string) ?? ""
+    : "";
```

Comment:
Don't get the reason for using redis here, we have all the past messages when we call this function, why not just grab them from there?

Reply:
Do we? We have the "nonSystemMessages" but the system prompt is generted each time from an array of actions